lib/std/os/linux/mips64.zig

```diff
@@ -14,7 +14,8 @@ const timespec = linux.timespec;
 pub fn syscall0(number: SYS) usize {
     return asm volatile (
         \\ syscall
-        \\ blez $7, 1f
+        \\ beq $7, $zero, 1f
+        \\ blez $2, 1f
         \\ dsubu $2, $0, $2
         \\ 1:
```

Comment:
The reason we don't have that is because musl mixes inline assembly and C logic: https://git.musl-libc.org/cgit/musl/tree/arch/mips64/syscall_arch.h#n38

We could of course do the same, but I thought I'd just keep things simple.

Reply:
it's not really related to C logic, it's just using C syntax to get values in and out of registers. Zig's constraint strings are powerful enough to do the same. You're already doing it with `={$2}` to get the value out. You can do the same thing with the input values.

```
	register long r4 __asm__("$4") = a;
```

is just 

```
: "[a] {$4}" (a),
```

in zig.